packages/app/src/overview/components/system/system.component.tsx

```diff
@@ -119,21 +122,27 @@ const System: FunctionComponent<Props> = ({
               <CardActionButton
                 active
                 labelMessage={messages.systemUpdateAction}
-                onClick={onUpdate}
+                // AUTO DISABLED - fix me if you like :)
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+                onClick={() => onUpdate()}
               />
             ) : (
               <CardActionButton
                 active
                 labelMessage={messages.systemDownloadAction}
-                onClick={onDownload}
+                // AUTO DISABLED - fix me if you like :)
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+                onClick={() => onDownload()}
                 data-testid={SystemTestIds.DownloadButton}
               />
             )
           ) : (
             <CardActionButton
               active
               labelMessage={messages.systemCheckForUpdates}
-              onClick={onUpdateCheck}
+              // AUTO DISABLED - fix me if you like :)
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+              onClick={() => onUpdateCheck()}
```

Comment:
Why this change with an arrow function is needed here? 🤔

Reply:
Under the hood, `onClick` was adding `event` to the function (which accepted optional parameter - Typescript was ok with that, but it was causing some unexpected problems). But you are right, this change seems suspicious. Decided to:
* revert the change and apply it higher
* provide small refactor of `ButtonTogglerItem` - now it provides TS types